### PR TITLE
Standardize Zowe Explorer configuration property names to make them consistent with VS Code's format

### PR DESCRIPTION
**Pull request by rudyflores**
## Proposed changes

- Review all Zowe Explorer configuration property names and make them consistent with VS Code's settings format for Zowe Explorer 2.0.
- Provide a way to convert Zowe Explorer settings that use the current format into the new standardized format (in order to make the user's transition smoother).
- Add standardization of configuration property names to the conformance criteria.
- Store the "Zowe-Temp-Folder-Location" setting as a string rather than an object.

## Release Notes

Release v2.0

Changelog:

- Standardized old configuration names to be under the dot and camel case standard for vscode
- Changed "Zowe-Temp-Folder-Location" schema from an array of strings to a string
- Change of configuration scopes for:
          - zowe.ds.default.* `resource`
          - zowe.ds.history `application`
          - zowe.uss.history `application`
          - zowe.jobs.history `application`
          - zowe.commands.history `application`

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [x] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

File: packages/zowe-explorer/src/extension.ts
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/extension.ts:154]: @rudyflores to keep activate() simpler to read and understand should we move this new section of code into a new file in the utils folder and call it from activate like checkConfigurations() or something?  @lauren-li what do you think?

Comment (lauren-li) [reply packages/zowe-explorer/src/extension.ts:154]: @JillieBeanSim @rudyflores Yes, I agree it should at least be a separate function! (Or maybe two functions? One for checking configuration version and one for doing the actual migration.) I think we had originally meant to separate it out from `activate()` once it was all working, but we got too excited once all the tests were finally passing and it slipped our minds.

**Comment by JillieBeanSim** [comment]
can also fix the audit errors in checks by updating like so in top-level package.json
image

File: packages/zowe-explorer/src/extension.ts
Comment (rudyflores) [reply packages/zowe-explorer/src/extension.ts:154]: I like this idea, this would help with maintaining the code :)

**Comment by rudyflores** [comment]
## Edge case regarding standardization (fixed)

An edge case was found when performing the migration of old settings, this issue was performed by doing the following steps:

1. Run the debug environment with user and workspace settings not being migrated, your user and workspace settings should not include any of the new standardized settings before performing the migration. Examples of these standardized settings can be any of the following:

- `zowe.ds.history`
- `zowe.uss.history`
- `zowe.jobs.history`
- `zowe.files.temporaryDownloadsFolder.path`

2. After the migration is performed (you should be able to see your old settings migrated into the new standardized settings) after the extension has activated successfully close the debug environment and remove those standardized settings only from the user settings. This way only the migrated settings will remain in the workspace settings and not in the user settings.

step 2


3. Once the standardized user settings are removed, run the extension development host again and the edge case should be replicated by showing how the settings are not migrated correctly for only the user settings specifically in all `zowe.*.history` based settings.

File: packages/zowe-explorer/src/globals.ts
Comment (JillieBeanSim) [inline packages/zowe-explorer/src/globals.ts:70]: if we are adding these 2 settings to the global constants, should all setting values be global constants as well or none as globals?  If the decision is all should be global setting then we would have to change all references to them in the code to call the global constant.

Comment (rudyflores) [reply packages/zowe-explorer/src/globals.ts:70]: Hmm if that would be the case on making them into global settings would that be out of scope from this item? I do remember @phaumer suggesting this change for the path and version since it would allow easier maintenance for user settings name changes in the future.

Comment (JillieBeanSim) [reply packages/zowe-explorer/src/globals.ts:70]: wouldn't `easier maintenance for user settings name changes in the future` need to cover all user setting names vs just 2 of them?

Comment (lauren-li) [reply packages/zowe-explorer/src/globals.ts:70]: I do think it might be a good idea for maintainability to add the other settings into the constants, as well. It would make sense to do as part of this PR on standardizing setting names.

Comment (rudyflores) [reply packages/zowe-explorer/src/globals.ts:70]: Sounds good then, I will go ahead and make these changes!

File: packages/zowe-explorer/src/utils/SettingsConfig.ts
Comment (lauren-li) [inline packages/zowe-explorer/src/utils/SettingsConfig.ts:55]: @rudyflores We might want to add more details to this message. Perhaps something like below:

`Settings have been successfully migrated for Zowe Explorer version 2 and above. To apply these settings, please reload your VS Code window.`

File: docs/Early Access - Changes Affecting Extenders.md
Comment (lauren-li) [inline docs/Early Access - Changes Affecting Extenders.md:49]: Use "VS Code".

File: docs/Early Access - Changes Affecting Extenders.md
Comment (lauren-li) [inline docs/Early Access - Changes Affecting Extenders.md:19]: Maybe add some reasoning here like, "These changes were made to conform to VS Code's [configuration schema](https://code.visualstudio.com/api/references/contribution-points#Configuration-schema)".

**Comment by JillieBeanSim** [comment]
@rudyflores @lauren-li that hang/need to click the button only happens the first time integration tests are ran and the settings are updated. Any time after that as long as the updated settings are in the settings.json it no longer hangs when integration tests are run.

**Review by JillieBeanSim — APPROVED** [review]

**Review by JillieBeanSim — APPROVED** [review]
Ha! dismissed my own review resolving conflicts with the merge from `next`,
LGTM

**Review by lauren-li — APPROVED** [review]
> @rudyflores @lauren-li that hang/need to click the button only happens the first time integration tests are ran and the settings are updated. Any time after that as long as the updated settings are in the settings.json it no longer hangs when integration tests are run.
> 
> since it happens on initialization it is hard to see where to put the automation of the click.

I think that hang happens during activation, before any tests start, so yes, I'm not sure how to automate/stub this, either. I was hoping maybe one of you would know. 😄 However, if not, I think we can just let the Zowe Explorer squad know about this when testing the `next` branch. Eventually (after everyone has migrated), it hopefully won't be an issue anymore.

I made a minor copyedit (sorry @JillieBeanSim, we'll need your re-approval again!), but other than that, I think this looks good. Thanks @rudyflores for addressing my comments, and thanks again for your patience and hard work on this PR!

**Review by JillieBeanSim — APPROVED** [review]

**Pull request merged**
